### ISSUE TITLE
Add EBS storage driver

### ISSUE THREAD

**Issue by clintonskitson** [emccode/libstorage#183]
This marks the introduction of the EC2 driver.

This driver is brought over from the existing driver and includes enhancements for:

- Updated and more granular tagging
- Retries on failed operations (handles API overloads)
- Updated next device discovery logic (handles phantom devices)
- Volume encryption option
- Locking mechanism on volume attach/detaches

**Comment by clintonskitson** [comment]
https://github.com/emccode/libstorage/issues/126

**Comment by clintonskitson** [comment]
This is to support Eucalyptus https://github.com/emccode/rexray/issues/431

**Comment by proudh** [comment]
Status update: I've refactored the code that was in REX-Ray 0.3.0's EC2 driver, and tested basic functionality with libStorage tests.  Currently I'm in the process of testing with REX-Ray; most of the commands work as expected, but I'm debugging a few stray ones.  I've also been working on adding in the enhancements listed in the issue; so far I've added in the lock for attaching/detaching volumes, and am close to finishing up implementing volume encryption.
